File: introductionToAlgorithmCLRS/chp2/Chapter_Problems/q3HornersRule.py
def HornersRule(degree, coef, x):
    val = 0
    for i in range(degree, -1, -1):
        val = coef[i] + x*val

    return val


if __name__ == "__main__":

    degree = int(input("Enter the degree of polynomial you wish to have : "))

    coef = list(map(lambda x: int(x), input("Enter the coefficients of the polynomial : ").split()))


    x = int(input("Enter the value of x : "))

    val = HornersRule(degree, coef, x);
    print("The calculated value is : " ,val)

File: introductionToAlgorithmCLRS/chp2/Chapter_Problems/q3HornersRule.c
#include <stdio.h>

int HornersRule(int degree, int coef[], int x) {
    int val = 0;
    for (int i = degree; i >-1; i--) {
        val = coef[i] + x*val;
    }

    return val;
}

void main() {

    printf("Enter the degree of polynomial you wish to have : ");
    int degree;
    scanf("%d", &degree);

    int coef[degree+1];
    printf("Enter the coefficients of the polynomial : ");

    for (int i = 0 ; i < degree+1; i++) {
        scanf("%d", &coef[i]);
    }

    printf("Enter the value of x : ");
    int x;
    scanf("%d", &x);
    int val = HornersRule(degree, coef, x);
printf("The calculated value is : %d" ,val);
}